### PR TITLE
Adding grant revoke admin endpoints

### PR DESCRIPTION
**Pull request by nidhi-kala**
Addressing #10222 

This PR will add API endpoints for granting and revoking admin rights.
I have added tests for the endpoints.

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

**Review by nijel — COMMENTED** [review]
To be consistent with other similar API endpoints, it should be:

* /admins/ endpoit - GET to list admins, POST to add
* /admins/(int:user_pk)/ - DELETE To delete

See how /components is handled at GroupViewSet.

**Comment by nidhi-kala** [comment]
Thank you for guiding me @nijel !
I have made the naming consistent for grant and revoke admin endpoints and updated the tests.

File: weblate/api/views.py [hunk 1]
```diff
@@ -631,6 +631,35 @@ def delete_components(self, request, id, component_id):
         obj.components.remove(component)
         return Response(status=HTTP_204_NO_CONTENT)
 
+    @action(detail=True, methods=["post"], url_path="admins")
+    def grant_admin(self, request, id):
+        group = self.get_object()
+        self.perm_check(request)
```
Comment (nijel) [inline weblate/api/views.py:637]: The perm_check should be extended to optionally accept object, so that group admins can also use this interface.

**Review by nijel — CHANGES_REQUESTED** [review]
Thanks, the code looks good, but there are few things which I'd like to see improved:

* The documentation is missing.
* The admins API will not work for group admins, see my comment on permission check.
* Can test for group admin doing changes be added?

**Pull request merged**